src/Quaternion.jl

```diff
@@ -13,8 +13,8 @@ Quaternion(s::Real, v1::Real, v2::Real, v3::Real, n::Bool = false) =
     Quaternion(promote(s, v1, v2, v3)..., n)
 Quaternion(x::Real) = Quaternion(x, zero(x), zero(x), zero(x), abs(x) == one(x))
 Quaternion(z::Complex) = Quaternion(z.re, z.im, zero(z.re), zero(z.re), abs(z) == one(z.re))
-Quaternion(s::Real, a::Vector) = Quaternion(s, a[1], a[2], a[3])
-Quaternion(a::Vector) = Quaternion(0, a[1], a[2], a[3])
+Quaternion(s::Real, a::AbstractVector) = Quaternion(s, a[1], a[2], a[3])
+Quaternion(a::AbstractVector) = Quaternion(0, a[1], a[2], a[3])
```

Comment:
If we add tests here, it would be like this:
```julia
@test Quaternion(1, [1,2,3]) === Quaternion(1,1,2,3)
@test Quaternion([1,2,3]) === Quaternion(0,1,2,3)
```
Personally, I feel this constructor is not useful because:

* It doesn't check the length of the argument.
* Accepting `Vector` makes performance degression. (cf. #61 )

So. I'd like to remove them in the next breaking release, but the removing should be done in another PR.

Reply:
> So. I'd like to remove them in the next breaking release, but the removing should be done in another PR.

Agreed. I think we should go further and make it so that every function that currently returns a vector returns a `Tuple` instead, and every function that accepts a vector can also accept a tuple. But yes, outside of this PR.

Reply:
Would you recommend adding
```julia
@test Quaternion(1, [1,2,3]) === Quaternion(1,1,2,3)
@test Quaternion([1,2,3]) === Quaternion(0,1,2,3)
```
to the test, or just leave it since there will be a future PR with a different constructor?
(Sry I wasn't sure if you are waiting for my response, or if this conversation is considered resolved)

Reply:
Sorry for the ambiguous comment. Yes, it seems better to have the tests I suggested. The tests will be useful when we will change the constructor behavior.

Reply:
Thanks for clarifying! (and sorry about the late reply)
I have rebased the PR branch and included the test, but since the test files have been largely changed, I am not sure it would match what the team was looking for.
Let me know if you have further comments, thanks!